snapcraft/internal/elf.py

```diff
@@ -114,6 +114,7 @@ def __init__(
         path: str,
         root_path: str,
         core_base_path: str,
+        content_dirs: Set[str],
```

Comment:
core_base_path and content_dirs could be generalized into one, can it not? It would need to stop being a set for content_dirs to be preferred over core_base_path.

Reply:
When doing this add a docstring explaining the importance of the order, then you do not need to do it in the three other places it is used.

You can make this type a Tuple instead of a set or list to not allow mutations.

I have not looked into even adding "root_path" to this, if it has more meanings than just a search path like the others, then keep it separate.

Reply:
Done.